examples/apps/cra-demo/README.md

```diff
@@ -1,215 +1,212 @@
-
-```tsx
-// inside useKVPair
-const [dataObject, setDataObject] = React.useState<KeyValueDataObject>();
-```
+React's `useEffect` is passed a function that fires as soon as the component loads and then fires again if any of its dependencies change (in our case the `dataObject` state variable).
 
-### 6.b Create/Load the Fluid container and data object
-Now that we have a setter method, we need to make sure that our create/get flow runs just once, on app load. Here we'll use React's `useEffect` because it allows code to be ran as soon as the component loads, and re-run only when specific values change, which by setting the dependency array to `[]`, means never.
+Our `useEffect` will end up firing twice. The first time, on component load, it will either create or load the container and `dataObject` based on if this is a new "document" (no hash in URL) or existing "document" (has hash in URL) .
 
-The bulk of this `useEffect` hooks is the async `load` function that starts by getting or creating the `fluidContainer`. The `FluidContainer` class then allows use to get or create one or more data objects. This could be multiple KVPairs, or other DataObjects that you define.
+> `TinyliciousService` is the service we are using to handle the Fluid data for this local demo. You'll use different services if you load your app in Azure, Teams or Sharepoint.
 
+The second time `useEffect` fires, the `dataObject` will have been set, so we set up a listener to call `updateData` each time the `changed` event is fired. `UpdateData` will sync the view state with the necessary Fluid data. For this example, we're just pulling all of the Fluid data.
 
 ```jsx
-// inside useKVPair
-React.useEffect(() => {
-    const { containerId, isNew } = getContainerId();
-
-    const load = async () => {
-        const fluidContainer = isNew
-            ? await Fluid.createContainer(containerId, [KeyValueDataObject])
-            : await Fluid.getContainer(containerId, [KeyValueDataObject]);
-
-        const keyValueDataObject: KeyValueDataObject = isNew
-            ? await fluidContainer.createDataObject(KeyValueDataObject, 'kvpairId')
-            : await fluidContainer.getDataObject('kvpairId');
-
-        setDataObject(keyValueDataObject);
-    }
-
-    load();
-
-}, [])
-```
-
-Once the DataObject is returned we assign it to our `dataObject` state variable, and now we have access to all of the KVPair's methods, including `set` which we pass down as the `setPair` function.
-
-Here's our hook so far.
-
+function App() {
 
-```tsx
-function useKVPair(): [KVData, SetKVPair | undefined] {
+    const [data, setData] = React.useState<{ [key: string]: any }>({});
     const [dataObject, setDataObject] = React.useState<KeyValueDataObject>();
 
     React.useEffect(() => {
-        const { containerId, isNew } = getContainerId();
-
-        const load = async () => {...
+        if (dataObject !== undefined) {
```

Comment:
Isn't this backwards?

```suggestion
        if (dataObject === undefined) {
```

Reply:
yeah, fixed it in one place, missed it here